core/tools.js

```diff
@@ -1950,6 +1950,40 @@
 				return ret;
 			},
 
+			/**
+			 * Returns the first element in the array for which the given callback `fn` returns true.
+			 * If there is no such element `undefined` is returned.
+			 *
+			 * ```javascript
+			 * var array = [ 'foo', 'bar', 'baz', 'faz' ];
+			 *
+			 * CKEDITOR.tools.array.find( array, function( item ) {
+			 *     return item.indexOf( 'az' ) > -1;
+			 * } ); // returns 'baz'.
+			 *
+			 * ```
+			 *
+			 * @param {Array} array An array to be iterated over.
+			 * @param {Function} fn A function with the signature `(a, index, array) -> b`.
```

Comment:
JSDuck allows you to explicitly specify callback (`fn` in this case) signature. You don't have to do it in a _custom_ way.

See https://github.com/senchalabs/jsduck/wiki/%40param#callback-parameters for more info.